packages/roadworks-cancelled-notification/index.ts

```diff
@@ -0,0 +1,172 @@
+                                             ><span style="padding-left: 5px;"
+                                               >GOV.UK</span
+                                             ></span
+                                           ></a
+                                         >
+                                       </div>
+                                     </div>
+                                   </header>
+                                  <div style="font-family: Arial, sans-serif; padding-left: 30px">
+                       <h1>Notification of cancelled Street Manager roadworks</h1>
+                       <p>There has been a cancelled roadwork update from Street manager. This may effect the end date showing in your disruption. Please review the roadwork and edit the disruption accordingly.The following disruptions are attached to cancelled roadworks;</p>
+                       ${disruptionLinks.map((link, i) => `<a href=${link}>Link to disruption ${i + 1}</a>`).join(" ")}
+                       <p>Amend the data accordingly, if needed.</p>
+                    </div>
+                                 </div>
+                               </div>
+                             </body>
+                           </html>`;
+
+export const createRoadworksCancellationEmail = (
+    contentsForEmail: {
+        disruptionIds: string[];
+        email: string;
+        orgId: string;
+    }[],
+    domainName: string,
+    stage: string,
+) => {
+    const domain = new URL(domainName);
+    const disruptionLinks = contentsForEmail.flatMap((content) =>
+        content.disruptionIds.map((disruptionId) => `${domain.toString()}/disruption-detail/${disruptionId}`),
+    );
+    const sourceEmail = isSandbox(stage) ? "no-reply@sandbox.cdd.dft-create-data.com" : `no-reply@${domain.hostname}`;
+
+    return new SendEmailCommand({
+        Destination: {
+            ToAddresses: contentsForEmail.map((content) => content.email),
+        },
+        Message: {
+            Body: {
+                Html: {
+                    Charset: "UTF-8",
+                    Data: roadworksCancellationEmailBody(disruptionLinks),
+                },
+            },
+            Subject: {
+                Charset: "UTF-8",
+                Data: "Notification of cancelled Street Manager roadworks",
+            },
+        },
+        Source: sourceEmail,
+    });
+};
+
+export const main = async (): Promise<void> => {
+    try {
+        logger.options.dev = process.env.NODE_ENV !== "production";
+        logger.options.debug = process.env.ENABLE_DEBUG_LOGS === "true" || process.env.NODE_ENV !== "production";
+
+        logger.options.meta = {
+            id: randomUUID(),
+        };
+
+        logger.info("Checking for cancelled roadworks...");
+
+        const {
+            DISRUPTIONS_TABLE_NAME: disruptionsTableName,
+            ORGANISATIONS_TABLE_NAME: orgTableName,
+            DOMAIN_NAME: domainName,
```

Comment:
We should check that `DOMAIN_NAME` and `STAGE` have been correctly set:

```
if (!disruptionsTableName || !orgTableName || !domainName || !stage) {
            throw new Error("Environment variables not correctly set.");
        }
```


Then remove the `|| ""` from lines 158 and 159.